### PR TITLE
Rename some 'sub_' functions

### PR DESCRIPTION
**Pull request by ecopsychologer**
I've added a few comments as I've been trying to improve readability of the code, but my main goal has been to rename the functions from 'sub_XXXXXXXX' to a descriptive name. On my machine the rom builds successfully and runs.

### CODE REVIEW

File: include/constants/pokemon.h [hunk 1]
```diff
@@ -63,7 +63,7 @@ enum PokemonDataParam {
     MON_DATA_3,
     MON_DATA_4,
     MON_DATA_SPECIES,
-    MON_DATA_HELD_ITEM,
+    MON_DATA_HELD_ITEM, // 6
```
Comment (lhearachel) [inline include/constants/pokemon.h:66]: remove these comments (and others)

File: include/overlay006/battle_params.h [hunk 2]
```diff
@@ -31,7 +31,7 @@ typedef struct {
     UnkStruct_0207D99C * unk_E4;
     UnkStruct_02026324 * unk_E8;
     PCBoxes * unk_EC;
-    UnkStruct_0202CC84 * unk_F0[4];
+    ChatotCryData * unk_F0[4];
```
Comment (lhearachel) [inline include/overlay006/battle_params.h:34]:
```suggestion
    ChatotCryData *chatotCry[MAX_BATTLERS];
```

File: include/overlay016/ov16_0223DF00.h [hunk 3]
```diff
@@ -196,7 +196,7 @@ int ov16_0223EDE0(BattleSystem * param0);
 u8 BattleSystem_TextSpeed(BattleSystem * param0);
 int BattleSystem_Ruleset(BattleSystem * param0);
 UnkStruct_02015F84 * ov16_0223EE28(BattleSystem * param0);
-UnkStruct_0202CC84 * BattleSystem_ChatotVoice(BattleSystem * param0, int param1);
+ChatotCryData * BattleSystem_ChatotVoice(BattleSystem * param0, int param1);
```
Comment (lhearachel) [inline include/overlay016/ov16_0223DF00.h:199]: Maybe rename this to `ChatotCry`?

File: include/overlay016/struct_ov16_02264408.h [hunk 4]
```diff
@@ -25,7 +25,7 @@ typedef struct {
     u32 unk_4C[4];
     int unk_50;
     UnkStruct_ov16_02264408_sub1 unk_54;
-    UnkStruct_0202CC84 * unk_6C;
+    ChatotCryData * unk_6C;
```
Comment (lhearachel) [inline include/overlay016/struct_ov16_02264408.h:28]:
```suggestion
    ChatotCryData *chatotCry;
```

(and elsewhere in the PR)

File: include/pokemon.h [hunk 5]
```diff
@@ -176,18 +176,20 @@ typedef struct BoxPokemon {
...
+    u32 status;
+    u8 level;
+    u8 mail;
+    u16 hp;
+    u16 maxHP;
+    u16 attack;
+    u16 defense;
+    u16 speed;
+    u16 spAtk;
+    u16 spDef;
```
Comment (lhearachel) [inline include/pokemon.h:192]: In general, maintain the Doxygen comments on struct fields (and add them, if possible).

File: include/pokemon.h [hunk 6]
```diff
@@ -176,18 +176,20 @@ typedef struct BoxPokemon {
...
-    u16 unk_06;
-    u16 unk_08;
-    u16 unk_0A;
-    u16 unk_0C;
-    u16 unk_0E;
-    u16 unk_10;
-    u16 unk_12;
+    u32 status;
+    u8 level;
+    u8 mail;
```
Comment (lhearachel) [inline include/pokemon.h:185]: Is this mail? Or an item?

File: include/pokemon.h [hunk 7]
```diff
@@ -196,8 +198,8 @@ typedef struct PartyPokemon {
  * @brief The main Pokemon data structure
  */
 typedef struct Pokemon {
-    BoxPokemon box; //!< Contains the pokemons boxed data
-    PartyPokemon party; //!< Contains the pokemons extra data while it is in the players party
+    BoxPokemon box; // Contains the pokemons boxed data
+    PartyPokemon party; // Contains the pokemons extra data while it is in the players party
```
Comment (lhearachel) [inline include/pokemon.h:202]: Keep the doxygen formatting on these comments

File: include/pokemon.h [hunk 8]
```diff
@@ -762,10 +764,10 @@ int Pokemon_LoadLevelUpMoveIdsOf(int monSpecies, int monForm, u16 *monLevelUpMov
 
 void Pokemon_ApplyPokerus(Party *party);
 u8 Pokemon_HasPokerus(Party *party, u8 param1);
-void sub_020777B4(Party *party, s32 param1);
+void UpdatePokerusStatusInParty(Party *party, s32 param1);
```
Comment (lhearachel) [inline include/pokemon.h:767]:
```suggestion
void Party_UpdatePokerusStatus(Party *party, s32 param1);
```

File: include/pokemon.h [hunk 9]
```diff
@@ -762,10 +764,10 @@ int Pokemon_LoadLevelUpMoveIdsOf(int monSpecies, int monForm, u16 *monLevelUpMov
...
 void Pokemon_ValidatePokerus(Party *party);
-BOOL sub_020778D8(Pokemon *mon);
-BOOL sub_020778F8(Pokemon *mon);
+BOOL IsPokemonInfectedWithPokerus(Pokemon *mon);
```
Comment (lhearachel) [inline include/pokemon.h:769]:
```suggestion
BOOL Pokemon_InfectedWithPokerus(Pokemon *mon);
```

File: include/pokemon.h [hunk 10]
```diff
@@ -762,10 +764,10 @@ int Pokemon_LoadLevelUpMoveIdsOf(int monSpecies, int monForm, u16 *monLevelUpMov
...
+BOOL CanPokemonSpreadPokerus(Pokemon *mon);
```
Comment (lhearachel) [inline include/pokemon.h:770]:
```suggestion
BOOL Pokemon_CanSpreadPokerus(Pokemon *mon);
```

File: include/pokemon.h [hunk 11]
```diff
@@ -884,14 +886,14 @@ BOOL Pokemon_SetRotomForm(Pokemon *mon, int monForm, int moveSlot);
...
-BOOL sub_02077E3C(Pokemon *mon);
-void sub_02077E64(Pokemon *mon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
-void sub_02077EE4(Pokemon *mon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
-void sub_02077F0C(Pokemon *mon, u32 param1, int param2);
+void HandleChatotSpecialInteraction(ChatotCryData *param0, int param1, u16 monSpecies, int param3, int param4, int param5, int param6, int param7);
+void HandleChatotSpecialAction(ChatotCryData *param0, int param1, u16 monSpecies, int param3, int param4, int param5, int param6, int param7, u8 param8);
+BOOL IsPokemonEligibleForAction(Pokemon *mon);
+void InitializePokemonAfterCapture(Pokemon *mon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
+void PostCapturePokemonProcessing(Pokemon *mon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
+void AssignHeldItemToPokemon(Pokemon *mon, u32 param1, int param2);
```
Comment (lhearachel) [inline include/pokemon.h:894]:
```suggestion
void Pokemon_HandleChatotInteraction(ChatotCryData *param0, int param1, u16 monSpecies, int param3, int param4, int param5, int param6, int param7);
void Pokemon_HandleChatotAction(ChatotCryData *param0, int param1, u16 monSpecies, int param3, int param4, int param5, int param6, int param7, u8 param8);
BOOL Pokemon_EligibleForAction(Pokemon *mon);
void Pokemon_InitAfterCatch(Pokemon *mon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
void Pokemon_UpdateAfterCatch(Pokemon *mon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
void Pokemon_GiveHeldItem(Pokemon *mon, u32 param1, int param2);
```

File: include/pokemon.h [hunk 12]
```diff
@@ -884,14 +886,14 @@ BOOL Pokemon_SetRotomForm(Pokemon *mon, int monForm, int moveSlot);
...
 BOOL Pokemon_CanLearnTM(Pokemon *mon, u8 tmID);
-BOOL Pokemon_CanFormLearnTM(u16 monSpecies, int monForm, u8 tmID);
+BOOL CanPokemonFormLearnTM(u16 monSpecies, int monForm, u8 tmID);
```
Comment (lhearachel) [inline include/pokemon.h:896]: Keep the `XXX_` naming scheme

File: include/struct_decls/struct_0202CC84_decl.h [hunk 13]
```diff
@@ -1,6 +1,6 @@
 #ifndef POKEPLATINUM_STRUCT_0202CC84_DECL_H
 #define POKEPLATINUM_STRUCT_0202CC84_DECL_H
 
-typedef struct UnkStruct_0202CC84_t UnkStruct_0202CC84;
+typedef struct ChatotCryData_t ChatotCryData;
```
Comment (lhearachel) [inline include/struct_decls/struct_0202CC84_decl.h:4]: Can you merge this into the struct-definition header? In general, we are trying to get rid of opaque structs.

File: include/unk_02006224.h [hunk 14]
```diff
@@ -5,17 +5,17 @@
...
+BOOL ProcessAudioInput(const ChatotCryData * param0, u32 param1, int param2, int param3);
+void ResetMicStatusFlags(void);
+MICResult StartMicSampling(void);
+MICResult StopMicSampling(void);
+void StoreMicDataInChatotCryStruct(ChatotCryData * param0);
+void SetMicProcessingFlag(u8 param0);
+BOOL ProcessChatotCryStructWithAudioParams(ChatotCryData * param0, u32 param1, int param2, int param3);
+BOOL ProcessChatotCryStructWithExtendedAudioParams(ChatotCryData * param0, u32 param1, int param2, int param3, u8 param4);
+int Sound_Chatter(ChatotCryData * param0);
+BOOL IsAudioParamValid(int param0);
```
Comment (lhearachel) [inline include/unk_02006224.h:19]: Probably give these `Sound` prefixes?

File: include/unk_0202CC64.h [hunk 15]
```diff
@@ -4,15 +4,15 @@
...
+int GetSizeOfChatotCryData(void);
+void ChatotCry_Init(ChatotCryData * param0);
+ChatotCryData * AllocateAndInitializeChatotCryData(int param0);
+ChatotCryData * GetChatotCryDataFromSave(SaveData * param0);
+BOOL IsChatotCryDataValid(const ChatotCryData * param0);
+void ResetChatotCryDataStatus(ChatotCryData * param0);
+const void * GetChatotCryAudioBuffer(const ChatotCryData * param0);
+void ProcessChatotCryAudioData(s8 * param0, const s8 * param1);
+void StoreProcessedAudioInChatotCryData(ChatotCryData * param0, const s8 * param1);
+void CopyChatotCryData(ChatotCryData * param0, const ChatotCryData * param1);
```
Comment (lhearachel) [inline include/unk_0202CC64.h:16]: `ChatotCry_` prefixes

File: include/struct_defs/struct_02090800.h [hunk 16]
```diff
@@ -12,11 +12,11 @@ typedef struct {
...
-    UnkStruct_02090800_sub1 unk_24;
-    UnkStruct_02090800_sub1 unk_2C;
-    UnkStruct_02090800_sub1 unk_34;
-} UnkStruct_02090800;
+    PokemonInfoDisplayStruct_sub1 unk_14;
+    PokemonInfoDisplayStruct_sub1 unk_1C;
+    PokemonInfoDisplayStruct_sub1 unk_24;
+    PokemonInfoDisplayStruct_sub1 unk_2C;
+    PokemonInfoDisplayStruct_sub1 unk_34;
+} PokemonInfoDisplayStruct;
```
Comment (lhearachel) [inline include/struct_defs/struct_02090800.h:20]:
```suggestion
    PokemonInfoDisplay_sub1 unk_14;
    PokemonInfoDisplay_sub1 unk_1C;
    PokemonInfoDisplay_sub1 unk_24;
    PokemonInfoDisplay_sub1 unk_2C;
    PokemonInfoDisplay_sub1 unk_34;
} PokemonInfoDisplay;
```

File: include/struct_defs/struct_02090800_sub1.h [hunk 17]
```diff
@@ -6,6 +6,6 @@
 typedef struct {
     int unk_00;
     Strbuf* unk_04;
-} UnkStruct_02090800_sub1;
+} PokemonInfoDisplayStruct_sub1;
```
Comment (lhearachel) [inline include/struct_defs/struct_02090800_sub1.h:9]:
```suggestion
} PokemonInfoDisplay_sub1;
```

Also, is there enough info on this struct to document the fields?

File: include/unk_02092494.h [hunk 18]
```diff
@@ -5,9 +5,9 @@
 #include "pokemon.h"
 #include "struct_defs/struct_02090800.h"
 
-UnkStruct_02090800 * sub_02092494(Pokemon * param0, BOOL param1, int param2);
-void sub_0209282C(UnkStruct_02090800 * param0);
+PokemonInfoDisplayStruct * sub_02092494(Pokemon * param0, BOOL param1, int param2);
+void sub_0209282C(PokemonInfoDisplayStruct * param0);
 void sub_0209304C(Pokemon * param0, TrainerInfo * param1, int param2, int param3, int param4);
-void sub_0209305C(BoxPokemon * param0, TrainerInfo * param1, int param2, int param3, int param4);
+void ProcessBoxPokemonWithTrainerInfo(BoxPokemon * param0, TrainerInfo * param1, int param2, int param3, int param4);
```
Comment (lhearachel) [inline include/unk_02092494.h:11]: What does this function do? `Process` is a bit generic.

File: src/pokemon.c [hunk 19]
```diff
@@ -123,11 +123,11 @@ static void BoxPokemon_ReplaceMove(BoxPokemon *boxMon, u16 moveID);
...
-static BOOL sub_020778E0(BoxPokemon *boxMon);
-static BOOL sub_02077900(BoxPokemon *boxMon);
-static void sub_02077EA4(BoxPokemon *boxMon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
-static void sub_02077EF8(BoxPokemon *boxMon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
-static BOOL BoxPokemon_CanLearnTM(BoxPokemon *boxMon, u8 tmID);
+static BOOL IsBoxPokemonInfectedWithPokerus(BoxPokemon *boxMon);
+static BOOL CanBoxPokemonSpreadPokerus(BoxPokemon *boxMon);
+static void InitializeBoxPokemonAfterCapture(BoxPokemon *boxMon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
+static void PostCaptureBoxPokemonProcessing(BoxPokemon *boxMon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
+static BOOL CanBoxPokemonLearnTM(BoxPokemon *boxMon, u8 tmID);
```
Comment (lhearachel) [inline src/pokemon.c:130]:
```suggestion
static BOOL BoxPokemon_InfectedWithPokerus(BoxPokemon *boxMon);
static BOOL BoxPokemon_CanSpreadPokerus(BoxPokemon *boxMon);
static void BoxPokemon_InitAfterCatch(BoxPokemon *boxMon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
static void BoxPokemon_UpdateAfterCatch(BoxPokemon *boxMon, TrainerInfo *param1, int monPokeball, int param3, int param4, int param5);
static BOOL BoxPokemon_LearnsTM(BoxPokemon *boxMon, u8 tmID);
```

File: src/unk_02006224.c [hunk 20]
```diff
@@ -11,45 +11,45 @@
...
+BOOL ProcessAudioInput(const ChatotCryData * param0, u32 param1, int param2, int param3);
+void ResetMicStatusFlags(void);
+MICResult StartMicSampling(void);
+MICResult StopMicSampling(void);
+void StoreMicDataInChatotCryStruct(ChatotCryData * param0);
+void SetMicProcessingFlag(u8 param0);
+BOOL ProcessChatotCryStructWithAudioParams(ChatotCryData * param0, u32 param1, int param2, int param3);
+BOOL ProcessChatotCryStructWithExtendedAudioParams(ChatotCryData * param0, u32 param1, int param2, int param3, u8 param4);
+int Sound_Chatter(ChatotCryData * param0);
+BOOL IsAudioParamValid(int param0);
```
Comment (lhearachel) [inline src/unk_02006224.c:25]: You can just delete these, their declarations should be included by the header.

File: src/unk_0202CC64.c [hunk 21]
```diff
@@ -7,51 +7,51 @@
 #include "unk_020244AC.h"
 #include "unk_0202CC64.h"
 
-typedef struct UnkStruct_0202CC84_t {
+typedef struct ChatotCryData_t {
     BOOL unk_00;
     s8 unk_04[1000];
-} UnkStruct_0202CC84;
+} ChatotCryData;
```
Comment (lhearachel) [inline src/unk_0202CC64.c:13]: Make this struct transparent.

File: src/unk_02092494.c [hunk 22]
```diff
@@ -13,25 +13,25 @@
 #include "pokemon.h"
```
Comment (lhearachel) [inline src/unk_02092494.c:13]: Is this file related to the Summary Screen?

**Review by lhearachel — CHANGES_REQUESTED** [review]
Thanks for your contributions! I've got some feedback to make sure that the function names match the general style of the repository.

File: include/overlay006/battle_params.h (hunk 2 above)
Comment (ecopsychologer) [reply include/overlay006/battle_params.h:34]: by renaming this though to chatotCry I believe we would have to rename every time something points to unk_F0 in this struct, and since the struct can be initiated with many names I'm not sure how to go about finding them all in a humane way

File: include/overlay016/ov16_0223DF00.h (hunk 3 above)
Comment (ecopsychologer) [reply include/overlay016/ov16_0223DF00.h]: done

File: include/pokemon.h (hunk 5 above)
Comment (ecopsychologer) [reply include/pokemon.h:192]: oh, I'm sorry I'm fairly new to this and thought that was an excited comment, I'll try to add some. What does it do?

Comment (ecopsychologer) [reply include/pokemon.h]: re-added there

File: include/unk_02092494.h (hunk 18 above)
Comment (ecopsychologer) [reply include/unk_02092494.h]: I updated it, it does kind of a bunch with cases but basically sets or processes the setting of box pokemon trainer info and met location

File: include/constants/pokemon.h (hunk 1 above)
Comment (ecopsychologer) [reply include/constants/pokemon.h:66]: done. I left the note about the data sanity because that might be helpful as people figure out what each piece of data does

File: include/pokemon.h (hunk 6 above)
Comment (ecopsychologer) [reply include/pokemon.h]: I noticed the data structure matched the previous games, but I believe it to be the ID of the mail the pokemon is holding

**Comment by ecopsychologer** [comment]
I'll look more at this later, thank you for reviewing!

File: include/overlay006/battle_params.h (hunk 2 above)
Comment (lhearachel) [reply include/overlay006/battle_params.h]: Fairy, I typically just treat it as necessary grunt work, but we can leave this until someone gets around to documenting the struct in its entirety.

File: include/pokemon.h (hunk 5 above)
Comment (lhearachel) [reply include/pokemon.h]: The Doxygen comments are for whenever we want to generate documentation, similar to Javadoc or Sphinx: https://www.doxygen.nl/

File: include/struct_defs/struct_02013A04_t.h [hunk 23]
```diff
@@ -1,7 +1,7 @@
 #ifndef POKEPLATINUM_STRUCT_02013A04_T_H
 #define POKEPLATINUM_STRUCT_02013A04_T_H
 
-struct UnkStruct_02013A04_t {
+struct ResourceMetadata_t {
```
Comment (lhearachel) [inline include/struct_defs/struct_02013A04_t.h:4]: Nitpicking on this one since it's short and a good example.

In an earlier file `include/struct_decls/struct_02013A04_decl.h`, we have this typedef:

```c
typedef struct ResourceMetadata_t ResourceMetadata;
```

The general style we're using for the gen4 projects is to merge these together and make the struct definition transparent. So, we'd instead have the single file with this:

```c
typedef struct ResourceMetadata {
    const void * unk_00;
    u32 unk_04;
} ResourceMetadata;
```

Then remove the includes for the previous file:

```c
#include "struct_decls/struct_02013A04_decl.h"
```

And, finally, delete the decl file.
